File: app/src/main/java/hide/demo/ClassTest3.java
package hide.demo;

/**
 * @author YvesCheung
 * 2018/9/18.
 *
 */
public class ClassTest3 {

    public void init() {

        /**
         * This usage is illegal
         * because {@link ClassTest3} is not a friend for {@link ClassNeedToHide}
         */
        ClassNeedToHide.INSTANCE.setDebuggable(true);
    }
}

File: app/src/main/java/usewith/demo/ClassTest3.kt
package usewith.demo

/**
 * @author YvesCheung
 * 2019-12-23
 */
class ClassTest3 {

    val a: UseWithCase by lazy {
        UseWithCase().also { it.init() }
    }

    fun deInit() {
        val b = a
        val c = b
        c.apply { deInit() }
    }
}